src/HDF5.jl

```diff
@@ -535,25 +536,33 @@ heuristic_chunk(x) = Int[]
 
 ### High-level interface ###
 # Open or create an HDF5 file
-function h5open(filename::AbstractString, rd::Bool, wr::Bool, cr::Bool, tr::Bool, ff::Bool,
+function h5open(filename::AbstractString, mode::String,
```

Comment:
I think you just explained better why I did it, I just put in a minimal docstring that covered the most common usage, but didn't cover all the optional arguments in any one method.